```
> Compilation failed; see the compiler output below.
  error: warnings found and -Werror specified
  /home/runner/work/kafka/kafka/storage/src/main/java/org/apache/kafka/storage/internals/log/LogCleaner.java:486: warning: [this-escape] possible 'this' escape before subclass is fully initialized
                      this::checkDone
                      ^
  1 error
  1 warning

```
It seems the suppression serves a purpose

--- a/storage/src/main/java/org/apache/kafka/storage/internals/log/LogCleaner.java
+++ b/storage/src/main/java/org/apache/kafka/storage/internals/log/LogCleaner.java
@@ -483,7 +483,9 @@ public class LogCleaner implements BrokerReconfigurable {
                     config.dedupeBufferLoadFactor,
                     throttler,
                     time,
-                    this::checkDone
+                    // Use a lambda instead of method reference to avoid `this` escape
+                    // The lambda captures `this` but doesn't evaluate it until runtime
+                    topicPartition -> checkDone(topicPartition)
             );
 
             if (config.dedupeBufferSize / config.numThreads > Integer.MAX_VALUE) {

